### ISSUE TITLE
Integration testing project set up - version solving failed.

### ISSUE THREAD

**Issue by EmanuelDavid** [flutter/flutter#90055]
pug get failed, because flutter_driver is not using latest platform sdk.

flutter_driver depends on platform 3.0.0 and open_mail_app 0.3.0 depends on platform ^3.0.2

Because every version of flutter_driver from sdk depends on platform 3.0.0 and open_mail_app 0.3.0 depends on platform ^3.0.2, flutter_driver from sdk is incompatible with open_mail_app 0.3.0.
And because no versions of open_mail_app match >0.3.0 <0.4.0, flutter_driver from sdk is incompatible with open_mail_app ^0.3.0.
So, because eo_charging depends on both open_mail_app ^0.3.0 and flutter_driver any from sdk, version solving failed.
pub get failed (1; So, because eo_charging depends on both open_mail_app ^0.3.0 and flutter_driver any from sdk, version solving failed.)

davidcondrea@Davids-iMac flutterApp % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.5.0, on macOS 11.5.2 20G95 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] Connected device (1 available)

• No issues found!
davidcondrea@Davids-iMac flutterApp %

**Comment by maheshmnj** [comment]
Hi @EmanuelDavid, this issue is because of the version conflicts of your dependencies in your `pubspec.yaml`  you need to use compatible versions to resolve the issue.

**Comment by EmanuelDavid** [comment]
So, let me try to explain again. I use a package named _open_mail_app_ that is up to date with last version of the package called _platform_, and that is _platform ^3.0.2_. BUT _flutter_driver_ depends on an older version of _platform_, and my assumption is that it should be updated.

**Comment by maheshmnj** [comment]
Could you please share your `pubspec.yaml` file 

Thanks

**Comment by EmanuelDavid** [comment]
description: A new Flutter application.

# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.6+1

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^1.0.2
  lint: ^1.5.2
  provider: ^6.0.0
  http: ^0.13.1
  flutter_secure_storage: ^4.1.0
  shared_preferences: ^2.0.5
  carousel_slider: ^4.0.0
  open_mail_app: ^0.3.0
  permission_handler: ^8.1.6
  flutter_switch: ^0.3.1
  webview_flutter: ^2.0.4
  qr_code_scanner: ^0.5.2
  wifi_iot: ^0.3.6
  flutter_blue: ^0.8.0
  flutter_spinkit: ^5.0.0
  wifi_info_flutter: ^2.0.0
  testfairy_flutter: any
  connectivity_plus: ^1.1.0
  geolocator: ^7.0.3
  google_maps_flutter: ^2.0.1
  geocoding: ^2.0.0
  share_plus: ^2.1.4
  persistent_bottom_nav_bar: ^4.0.2
  url_launcher: ^6.0.3
  back_button_interceptor: ^5.0.0
  intl: ^0.17.0
  easy_debounce: ^2.0.0
  device_info_plus: ^2.1.0
  throttling: ^1.0.0

dev_dependencies:
  flutter_test:
    sdk: flutter

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  assets:
    - assets/images/
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  fonts:
    - family: Circular-Book
      fonts:
        - asset: assets/fonts/CircularPro-Book.otf
          weight: 400
    - family: Circular-Black
      fonts:
        - asset: assets/fonts/CircularPro-Black.otf
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

**Comment by maheshmnj** [comment]
@EmanuelDavid I added all your plugins in a sample flutter project and `flutter pub get` finished with `exit code 0` (with no issues) could you please share the complete output of `flutter doctor -v`.

Thanks
